### DISCUSSION TITLE
Automatic range of the colorbar – Data visualization

### DISCUSSION THREAD

**Discussion by grkal** [groops-devs/groops#159] [Q&A] [answered]
Dear GROOPS Team,

The content of this message concerns the colorbar range corresponding to the eclipse factors (EFs) of GRACE/GRACE-FO when trying to plot them as colorbar values in a plot including the corresponding argument of latitudes with respect to time.

As an example, I will use the GRACE-C EFs and the corresponding argument of latitudes from 2019-04-01 in an attempt to produce a similar plot for the aforementioned date with the one for the whole month of April, 2019 displayed in the GROOPS documentation (section “Data visualization”, subchapter “Instrument data handling”, chapter “Cookbook”).

Attached you will find the “ColorbarRangeEFsDataVisualization.zip” file which includes the GRACE-C GNI1B file for the test date [2019-04-01] already converted in GROOPS format [GNI1B_2019-04-01_C_04_GROOPS.txt] and the “ColorbarRangeEFsDataVisualization.xml” file which is a script I created leading to plots of the EFs as colorbar values with respect to the argument of latitude for the test date.

As you will see when executing the aforementioned script, when the “PlotGraph” program is run with no limits set for the colorbar range, the following message appears with respect to the colorbar:

[_plot z range  (-1.85602 .. 1.85602) of (-1.85602 .. 1.85602)_]

However, the expected range of the EFs is between 0 and 1 which is confirmed by the “InstrumentArcStatistics” program also included in the script.

In order to properly plot the colorbar values, I have to manually set the values of the “max” and “min” config elements to “1” and “0” respectively as you will also see after executing the attached script.

Is there a specific reason why the colorbar range is different from the expected one when no limits are used or am I doing something wrong when trying to produce the plot?

Thank you very much in advance for your time and looking forward to your feedback.

Best regards,
Grigorios Kalimeris
Chair of Astronomical and Physical Geodesy
Technical University of Munich
[ColorbarRangeEFsDataVisualization.zip](https://github.com/groops-devs/groops/files/10834939/ColorbarRangeEFsDataVisualization.zip)

**Comment by tmayerguerr** [comment, marked as answer]
Dear Grigorios,

If the minimum and maximum values are not specified, GROOPS tries to find a good range automatically. Since we have found that the minimum and maximum values of the data set do not give good results for data that is approximately normally distributed, we have decided to set the color bar to three times the mean absolute value. This works well for geoid, gravity anomalies and measurement residuals. Of course, you can set your own reasonable limits.

regards
Torsten

**Comment by grkal** [comment]
Dear Torsten,

Thank you very much about your detailed reply regarding the way the colorbar values are created in GROOPS.

Best regards,
Greg
